store/tooling/tooling.go

```diff
@@ -346,25 +403,54 @@ func (tsto *ToolingStore) DownloadMany(toDownload []SnapToDownload, curSnaps []*
 		})
 	}
 
-	sars, _, err := tsto.sto.SnapAction(context.TODO(), current, actions, nil, nil, nil)
+	sars, _, err := tsto.sto.SnapAction(context.TODO(), current, actions, nil, nil,
+		&store.RefreshOptions{IncludeResources: true})
 	if err != nil {
 		// err will be 'cannot download snap "foo": <reasons>'
-		return nil, err
+		return nil, nil, err
 	}
 
-	for _, sar := range sars {
-		targetPath, err := opts.BeforeDownloadFunc(sar.Info)
+	downloadedComps = make(map[string][]*DownloadedComponent, len(sars))
+	for i, sar := range sars {
+		// Create component infos from resource data for the components we will download
+		cinfos := make(map[string]*snap.ComponentInfo, len(sar.Resources))
+		for _, res := range sar.Resources {
+			if !strutil.ListContains(toDownload[i].CompsToDownload, res.Name) {
```

Comment:
Should we do something other than skip here? Although I guess skipping might be good under some conditions, since snaps can lose components.

Maybe log a message, not sure what our usual habits are around logging are, tbh.

Reply:
If we do continue to skip requested components that don't exist as resources, we should add a test for that.

Reply:
Afaiu `SnapAction` returns information about all resources for the specified snap, or at least I do not see how Components are passed to it otherwise. So here we are filtering out resources we do not have interest in.